### PR TITLE
feat: Automatic https support with Caddy

### PR DESCRIPTION
**Pull request by jakubigla**

### CODE REVIEW

File: variables.tf [hunk 1]
```diff
@@ -9,28 +9,42 @@ variable "location" {
...
-  description = "The required memory of the Atlantis container in GB"
-  type        = number
-  default     = 2
-}
-
-variable "port" {
-  description = "Port on which Atlantis is listening"
-  type        = number
-  default     = 4141
+variable "atlantis_container" {
```
Comment (dgniewek) [inline variables.tf:12]: Maybe this variable should be renamed? 
since description doesn't match the variable name IMO

File: variables.tf [hunk 2]
```diff
@@ -234,27 +248,50 @@ variable "repo_config_file" {
...
-
-variable "secure_environment_variables" {
-  description = "A list of sensitive environment variables to be set on the container"
-  type        = map(string)
-  default     = {}
-}
-
-variable "secure_environment_variables_from_key_vault" {
-  description = "A list of sensitive environment variables to be set on the container read from Azure Key Vault"
+variable "containers" {
```
Comment (dgniewek) [inline variables.tf:251]: What's the difference between that variable and `atlantis_container`?

File: modules/caddy-automatic-https/locals.tf [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+  name_from_descriptor = trim(replace(
+    lookup(module.this.descriptors, "azure-container-group", module.this.id),
+    "/${module.this.delimiter}${module.this.delimiter}+/",
+  module.this.delimiter), module.this.delimiter)
+
+  location            = coalesce(one(data.azurerm_resource_group.this[*].location), var.location)
+  resource_group_name = coalesce(one(data.azurerm_resource_group.this[*].name), var.resource_group_name)
+
+  dns_name_label = (length(var.subnet_ids) == 0
+    ? (var.dns_name_label != null ? var.dns_name_label : local.name_from_descriptor)
```
Comment (dgniewek) [inline modules/caddy-automatic-https/locals.tf:13]: Minor - I don;t like that linting - `?` in the new line
Maybe few more new lined would make it more clear

File: variables.tf (hunk 2 above)
Comment (jakubigla) [reply variables.tf:251]: it gives you flexibility to add your own sidecar containers should you want them deploy along the atlantis one

**Review by Santhin — APPROVED** [review]

File: modules/caddy-automatic-https/main.tf [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+
+  name = var.resource_group_name
+}
+
+module "caddy_persistence_storage_account" {
+  source  = "getindata/storage-account/azurerm"
+  version = "1.3.0"
+  context = module.this.context
+
+  enabled = var.caddy_persistence_storage_account == null
```
Comment (dgniewek) [inline modules/caddy-automatic-https/main.tf:12]: :exploding_head:

File: modules/caddy-automatic-https/templates/Caddyfile.tpl [hunk 5]
```diff
@@ -0,0 +1,3 @@
+${hostname}
+
+reverse_proxy localhost:4141
```
Comment (dgniewek) [inline modules/caddy-automatic-https/templates/Caddyfile.tpl:3]: Maybe we should also template the port - I can imagine a use case where some other side-container is using `4141`

**Review by dgniewek — APPROVED** [review]

**Review by dgniewek — APPROVED** [review]

**Review by dgniewek — APPROVED** [review]
LGTMT

**Pull request merged**